src/pages/LearnPage.tsx

```diff
@@ -64,101 +59,96 @@ function LessonGroup({ title, children }: LessonGroupProps) {
-  const [lesson, setLesson] = useState<React.ReactNode>(null)
+  const groupedLessons: {
+    [key: string]: lesson[]
+  } = {}
+
+  lessons.forEach((lesson) => {
+    if (!(lesson.group in groupedLessons)) {
+      groupedLessons[lesson.group] = []
+    }
+
+    groupedLessons[lesson.group].push(lesson)
+  })
 
-  return !lesson ? (
+  return (
     <Container sx={{ p: 15 }}>
       <Typography variant="h3" gutterBottom color="primary">
         Learn
       </Typography>
 
       <SearchBar />
 
-      <LessonGroup title="Intro Algo Design">
-        <InlineSpacing spacing={40} />
-
-        <LessonButton
-          onClick={() =>
-            setLesson(
-              <LessonPage
-                blocks={[
-                  <ReadingBlock content={dp} />,
-                  <MCQBlock
-                    question={q1}
-                    answerOptions={ans1}
-                    correctOptions={correct1}
-                  />,
-                  <MCQBlock
-                    question={q2}
-                    answerOptions={ans2}
-                    correctOptions={correct2}
-                  />,
-                  <FITBBlock
-                    question={
-                      'Is this the real life? Is this just {fantasy}? ' +
-                      'Caught in a {landslide}, no escape from reality. ' +
-                      'Open your eyes, look up to the {skies} and see...'
-                    }
-                  />,
-                  <ReadingBlock content={markdown1} />,
-                  <ReadingBlock content={markdown2} />,
-                  <ReadingBlock content={markdown3} />,
-                ]}
-              />,
-            )
-          }
-        >
-          Brute Force
-        </LessonButton>
-
-        <InlineSpacing spacing={60} />
-
-        <LessonButton>Intro Greedy</LessonButton>
-
-        <InlineSpacing spacing={60} />
-
-        <LessonButton>Time & Memory Analysis</LessonButton>
-      </LessonGroup>
-
-      <Divider />
-
-      <LessonGroup title="Intro Data Structures">
-        <InlineSpacing spacing={40} />
-
-        <LessonButton>Prefix Sums</LessonButton>
-
-        <InlineSpacing spacing={60} />
-
-        <LessonButton>Lists & Vectors</LessonButton>
-
-        <InlineSpacing spacing={60} />
-
-        <LessonButton>Stacks & Queues</LessonButton>
-
-        <InlineSpacing spacing={40} />
-
-        <NextButton>
-          <ArrowForwardIcon style={{ fontSize: '3rem' }} />
-        </NextButton>
-      </LessonGroup>
+      {Object.entries(groupedLessons).map((lessonGroup, groupIndex) => {
+        return (
+          <>
+            <LessonGroup title={lessonGroup[0].toString()}>
+              <InlineSpacing spacing={40} />
+              {lessonGroup[1].map((lesson) => (
+                <>
+                  <LessonButton>
+                    <Link
+                      style={{ color: 'inherit', textDecoration: 'none' }}
+                      to={'/learn/' + lesson.urlParam}
```

Comment:
this is a super nit (I'm not picky about this), but some people prefer to use formatted strings like 
\`/learn/${lesson.urlParam}\`

Reply:
good to know! i'm going to leave it though since i don't think it makes a difference in this case